xdsl/transforms/riscv_register_allocation.py

```diff
@@ -4,18 +4,31 @@
 from xdsl.passes import ModulePass
 
 
+def allocate_registers(op: ModuleOp) -> None:
```

Comment:
Splitting this function out made it a bit easier to test in the `workshop-all` branch, compared with the admittedly tiny overhead of creating the ModulePass and context and calling apply on that. Happy to revert